src/librustdoc/config.rs

```diff
@@ -469,17 +469,7 @@ impl Options {
             }
         }
 
-        let edition = if let Some(e) = matches.opt_str("edition") {
-            match e.parse() {
-                Ok(e) => e,
-                Err(_) => {
-                    diag.struct_err("could not parse edition").emit();
-                    return Err(1);
-                }
-            }
-        } else {
-            DEFAULT_EDITION
-        };
```

Comment:
Oh oof, does that mean rustdoc was treating --edition=2021 as --edition=2015 before? :(

Reply:
I don't think so. The 2015 default was only if `--edition` isn't passed.